features/netsocket/emac-drivers/TARGET_Cypress/TARGET_PSOC6/TARGET_WHD/interface/whd_emac.cpp

```diff
@@ -0,0 +1,205 @@
+{
+}
+
+WHD_EMAC::WHD_EMAC()
+    : interface_type(WHD_STA_ROLE)
+{
+}
+
+WHD_EMAC &WHD_EMAC::get_instance() {
+    return get_instance(WHD_STA_ROLE);
+}
+
+WHD_EMAC &WHD_EMAC::get_instance(whd_interface_role_t role) {
+    static WHD_EMAC emac_sta(WHD_STA_ROLE);
+    static WHD_EMAC emac_ap(WHD_AP_ROLE);
+    return role == WHD_AP_ROLE ? emac_ap : emac_sta;
+}
+
+uint32_t WHD_EMAC::get_mtu_size() const
+{
+    return WHD_PAYLOAD_MTU;
+}
+
+uint32_t WHD_EMAC::get_align_preference() const
+{
+    return 0;
+}
+
+void WHD_EMAC::add_multicast_group(const uint8_t *addr)
+{
+    memcpy(multicast_addr.octet, addr, sizeof(multicast_addr.octet));
+    whd_wifi_register_multicast_address(ifp, &multicast_addr);
+}
+
+void WHD_EMAC::remove_multicast_group(const uint8_t *addr)
+{
+    whd_wifi_unregister_multicast_address(ifp, &multicast_addr);
+}
+
+void WHD_EMAC::set_all_multicast(bool all)
+{
+    //TODO: new API call for this in the whd interface to be added.
+    //whd_wifi_set_iovar_value(ifp, IOVAR_STR_ALLMULTI, all);
+}
+
+void WHD_EMAC::power_down()
+{
+    powered_up = false;
+    whd_wifi_off(ifp);
+    whd_deinit(ifp);
+}
+
+bool WHD_EMAC::power_up()
+{
+	drvp = *(cybsp_get_wifi_driver());
+	whd_wifi_on(drvp, &ifp /* OUT */);
+    powered_up = true;
+    if (link_state && emac_link_state_cb ) {
+        emac_link_state_cb(link_state);
+    }
+    return true;
+}
+
+bool WHD_EMAC::get_hwaddr(uint8_t *addr) const
+{
+    whd_mac_t mac;
+    whd_result_t res = whd_wifi_get_mac_address(ifp, &mac);
+    MBED_ASSERT(res == WHD_SUCCESS);
+    memcpy(addr, mac.octet, sizeof(mac.octet));
+    return true;
+}
+
+void WHD_EMAC::set_hwaddr(const uint8_t *addr)
+{
+  /* No-op at this stage */
+}
+
+uint8_t WHD_EMAC::get_hwaddr_size() const
+{
+    whd_mac_t mac;
+    return sizeof(mac.octet);
+}
+
+void WHD_EMAC::set_link_input_cb(emac_link_input_cb_t input_cb)
+{
+    emac_link_input_cb = input_cb;
+}
+
+void WHD_EMAC::set_link_state_cb(emac_link_state_change_cb_t state_cb)
+{
+    emac_link_state_cb = state_cb;
+}
+
+void WHD_EMAC::set_memory_manager(EMACMemoryManager &mem_mngr)
+{
+    memory_manager = &mem_mngr;
+}
+
+bool WHD_EMAC::link_out(emac_mem_buf_t *buf)
+{
+    uint16_t offset = 64;
+    whd_buffer_t buffer;
+
+    uint16_t size = memory_manager->get_total_len(buf);
+
+    whd_result_t res = whd_host_buffer_get(drvp, &buffer, WHD_NETWORK_TX, size+offset, WHD_TRUE);
+    MBED_ASSERT(res == WHD_SUCCESS);
+
+    whd_buffer_add_remove_at_front(drvp, &buffer, offset);
+
+    void *dest = whd_buffer_get_current_piece_data_pointer(drvp, buffer);
+    memory_manager->copy_from_buf(dest, size, buf);
+
+    whd_network_send_ethernet_data(ifp, buffer);
+    memory_manager->free(buf);
+    return true;
+}
+
+void WHD_EMAC::get_ifname(char *name, uint8_t size) const
+{
+    memcpy(name, "wl", size);
+}
+
+extern "C"
+{
+void cy_network_process_ethernet_data(whd_interface_t ifp, whd_buffer_t buffer)
+{
+	emac_mem_buf_t *mem_buf = NULL;
```

Comment:
alignment issue here - pls run astyle with our coding style file (.astylerc)

Reply:
Fixed by applying astyle in https://github.com/ARMmbed/mbed-os/pull/10692/commits/c115f23028c674329e859c26f4b7ad06f5d3d66c#diff-38f33b303ec6a94c6f64f1bd8778ca90R162